### DISCUSSION TITLE
CSV-import in Kimai Cloud?

### DISCUSSION THREAD

**Discussion by BuzzDausBerlin** [kimai/kimai#3919] [General]
Hi, will there be the option to import data (specifically time data) in the near future?

**Comment by kevinpapst** [comment]
There is already the option. See System > Importer

**Reply by BuzzDausBerlin** [reply]
Hi, thanks for the quick reply. 
I've tried the importer but I get Server error 500 when "project" and "activity" are not empty. When they are empty, I get a regular error, that the fields are missing

**Reply by kevinpapst** [reply]
> Failed to parse time string (44927 11:45) at position 4 (7): Unexpected character

Why do you need the importer? Moving to the Cloud from a self-hosted environment?

**Reply by BuzzDausBerlin** [reply]
Yes. There's not many data that needs to be imported (only for 2023) and I'm using the migration to restructure some things. Still, importing part of the data would be useful

**Reply by kevinpapst** [reply]
Write me an email. I can import the database dump.

**Reply by BuzzDausBerlin** [reply]
I don't want that because I want to make some changes regarding activities. Is the error when importing a known bug already?

**Reply by kevinpapst** [reply]
There is no bug (well. except that the error is no properly catched). 
The importer works, but it expects the data in a certain way. 
See the demo files that can be downloaded from the page. 

I posted the error above: 

> Failed to parse time string (44927 11:45) at position 4 (7): Unexpected character

**Reply by BuzzDausBerlin** [reply]
I see, thanks. Didn't see the error above, sorry. Ok, will change the date accordingly

**Reply by BuzzDausBerlin** [reply]
Ok, it works now. 
Only problem is: event though there's an activity with the exact same name, a new one is created (that has no customer or project). Will be no big deal for me, I can bulk edit the entries and delete the activities afterwards. But maybe this can be fixed some time? Or is it me again and I'm making another mistake in the CSV?

**Reply by kevinpapst** [reply]
When I clone the same row 10 times, I get one activity.
When I change three customer names of these 10 rows, I get one activity.
Share your CSV and I will test

**Reply by BuzzDausBerlin** [reply]
sent
